nomad/state/state_store.go

```diff
@@ -2543,7 +2543,7 @@ func (s *StateStore) updateDeploymentWithAlloc(index uint64, alloc, existing *st
 		return err
 	}
 	if deployment == nil {
-		return fmt.Errorf("allocation %q references unknown deployment %q", alloc.ID, alloc.DeploymentID)
```

Comment:
Nah because it is somewhat expected (though rarely) and isn't actually an error.